src/main/java/org/threadly/examples/features/PrioritySchedulerExample.java

```diff
@@ -0,0 +1,164 @@
+package org.threadly.examples.features;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import org.threadly.concurrent.PriorityScheduler;
+import org.threadly.concurrent.TaskPriority;
+import org.threadly.concurrent.future.FutureUtils;
+import org.threadly.concurrent.future.ListenableFuture;
+
+/**
+ * Provides a basic example of the PriorityScheduler feature.
+ * 
+ * @author sreid8 - Sean Reid
+ *
+ */
+public class PrioritySchedulerExample {
+  
+  private static Boolean threadDetail = false;
+  
+  /**
+   * Main entry point of the example.
+   * @param args -> program args. run w/o args for help
+   */
+  public static void main(String[] args) {
+    if (!handleArgs(args)) {
+      System.exit(1);
+    }
+    
+    final int numThreads = Integer.parseInt(args[0]);
+    
+    //create the PriorityScheduler. This is where we will make a method to handle the case of different
+    //priorities!!!
+    final PriorityScheduler executor = new PriorityScheduler(numThreads - 1, true);
+    
+    //create a list of ListenableFutures (java.util.concurrent.Future + the ability to listen for
+    //completion of task) with the number of threads in the thread pool.
+    List<ListenableFuture<?>> futures = new ArrayList<ListenableFuture<?>>(numThreads - 1);
+    
+    //add numThreads - 1 tasks to the PriorityScheduler.
+   for (int i = 0; i <= numThreads; i++) {
+     //random between 0-2 to determine priority.
+     TaskPriority priority = getRandomPriority();
```

Comment:
I was thinking in these examples we would do things which are more likely to be real use cases.  I can't think of a use case where a random priority would be helpful.  So for example I would recommend this as an alternative:

Submit some tasks with priority, but have no implementation, instead just a comment describing what they would do.  Either as inner classes, or anonymous inner classes, some examples we may want to show:
* A recurring low priority task that checks a source for configuration...we could describe how this task could be checking for a configured pool size, and updating the pool size at runtime if desired
* A recurring low priority task that does some cleanup work...possibly a cache cleaner, or something similar (again no implementation, just a comment describing a use case)...it might also be worth constructing the priority scheduler with a longer non-default low priority wait time, and describe that (in our silly example) low priority tasks can wait for say 5 or 10 seconds beyond scheduled run time without any impact in the system
* A recurring high priority task that reports statistics.  One possible stat it could be reporting is how many tasks are queued in the pool, and how many tasks are running.
* Maybe have some high priority executions, which could either come in from a public function (that is never invoked), or something like this loop